cg/cli/workflow/balsamic/base.py

```diff
@@ -0,0 +1,335 @@
+            tumor_paths.add(concatenated_paths[1])
+        else:
+            normal_paths.add(concatenated_paths[1])
+
+        if link_obj.sample.bed_version:
+            target_beds.add(link_obj.sample.bed_version.filename)
+
+    nr_paths = len(tumor_paths) if tumor_paths else 0
+    if nr_paths != 1:
+        click.echo(
+            f"Must have exactly one tumor sample! Found {nr_paths} samples.",
+            color="red",
+        )
+        context.abort()
+    tumor_path = tumor_paths.pop()
+
+    normal_path = None
+    nr_normal_paths = len(normal_paths) if normal_paths else 0
+    if nr_normal_paths > 1:
+        click.echo(f"Too many normal samples found: {nr_normal_paths}", color="red")
+        context.abort()
+    elif nr_normal_paths == 1:
+        normal_path = normal_paths.pop()
+
+    # Call Balsamic
+    command_str = (
+        f" config case"
+        f" --reference-config {reference_config}"
+        f" --singularity {singularity}"
+        f" --tumor {tumor_path}"
+        f" --case-id {case_id}"
+        f" --output-config {case_id}.json"
+        f" --analysis-dir {root_dir}"
+        f" --umi-trim-length {umi_trim_length}"
+    )
+    if target_bed:
+        command_str += f" -p {target_bed}"
+    elif len(target_beds) == 1:
+        bed_path = Path(context.obj["bed_path"])
+        command_str += f" -p {bed_path / target_beds.pop()}"
+    else:
+        raise BalsamicStartError("No target bed specified!")
+
+    if normal_path:
+        command_str += f" --normal {normal_path}"
+    if umi:
+        command_str += f" --umi"
+    if quality_trim:
+        command_str += f" --quality-trim"
+    if adapter_trim:
+        command_str += f" --adapter-trim"
+    command = [f"bash -c 'source activate {conda_env}; balsamic"]
+    command_str += "'"  # add ending quote from above line
+    command.extend(command_str.split(" "))
+    if dry:
+        click.echo(" ".join(command))
+        return SUCCESS
+    else:
+        process = subprocess.run(" ".join(command), shell=True)
+        return process
+
+
+@balsamic.command()
+@click.option("-d", "--dry-run", "dry", is_flag=True, help="print command to console")
+@click.option(
+    "-r",
+    "--run-analysis",
+    "run_analysis",
+    is_flag=True,
+    default=False,
+    help="start " "analysis",
+)
+@click.option("--config", "config_path", required=False, help="Optional")
+@PRIORITY_OPTION
+@EMAIL_OPTION
+@click.argument("case_id")
+@click.pass_context
+def run(context, dry, run_analysis, config_path, priority, email, case_id):
+    """Generate a config for the case_id."""
+
+    conda_env = context.obj["balsamic"]["conda_env"]
+    slurm_account = context.obj["balsamic"]["slurm"]["account"]
+    priority = priority if priority else context.obj["balsamic"]["slurm"]["qos"]
+    root_dir = Path(context.obj["balsamic"]["root"])
+    if not config_path:
+        config_path = Path.joinpath(root_dir, case_id, case_id + ".json")
+
+    # Call Balsamic
```

Comment:
Why is there no method in the balsamic AnalysisAPI for running balsamic? I think it would make more sense calling balsamic from within the API.

Reply:
Sure. What benefit do you see?

Reply:
I'm thinking that is the point of abstracting each tool to their own api classes to begin with. i.e. all functionality of a certain tool is handled at one place only.

Reply:
That would be nice, but let's do that refactoring in a future PR